compiler/src/main/scala/org/scalajs/core/compiler/JSEncoding.scala

```diff
@@ -61,7 +61,7 @@ trait JSEncoding extends SubComponent { self: GenJSCode =>
       longName = base+"$"+suffix
     }
     usedLocalNames += longName
-    longName
+    mangleJSName(longName)
```

Comment:
I almost said "yes, we can do that". But he problem is the `OrReserved`, i.e., `"arguments"`, `"eval"` and `ScalaJSEnvironmentName == "ScalaJS"`. These are not taken into account by `mangleJSName`. Note that we cannot "fix" `mangleJSName` because it defines how public names such as methods are mangled, so changing the algorithm of `mangleJSName` breaks binary compatibility.

I could change `isKeywordOrReserved(longName)` into `isReserved(longName)`, though.